tests/e2e/ans/ans.test.ts

```diff
@@ -0,0 +1,168 @@
+      expect(isValidANSName("secondary.primary.apt")).toEqual({ domainName: "primary", subdomainName: "secondary" });
+    });
+
+    test("it returns false for invalid names", () => {
+      expect(() => isValidANSName("")).toThrow();
+      expect(() => isValidANSName(".")).toThrow();
+      expect(() => isValidANSName("..")).toThrow();
+      expect(() => isValidANSName(" . ")).toThrow();
+      expect(() => isValidANSName(" test ")).toThrow();
+      expect(() => isValidANSName(".apt")).toThrow();
+      expect(() => isValidANSName(".apt.apt")).toThrow();
+      expect(() => isValidANSName(".apt.")).toThrow();
+      expect(() => isValidANSName("1")).toThrow();
+      expect(() => isValidANSName("1.apt")).toThrow();
+      expect(() => isValidANSName("bad.bad.bad")).toThrow();
+    });
+  });
+
+  describe("registerName", () => {
+    test("can be called with a variety of parameters", async () => {
+      const alice = Account.generate();
+      await aptos.fundAccount({
+        accountAddress: alice.accountAddress.toString(),
+        amount: 500_000_000,
+      });
+      const name = randomString();
+
+      expect(
+        await aptos.registerName({
+          name,
+          sender: alice,
+          expiration: { policy: "domain", years: 1 },
+        }),
+      ).toBeTruthy();
+
+      await expect(
+        aptos.registerName({
+          sender: alice,
+          name,
+          // Force the year to be absent
+          expiration: { policy: "domain" } as any,
+        }),
+      ).rejects.toThrow();
+
+      // Testing to make sure that the subdomain policy is enforced
+      await expect(
+        aptos.registerName({
+          sender: alice,
+          name,
+          // Force the year to be absent
+          expiration: { policy: "subdomain:follow-domain" },
+        }),
+      ).rejects.toThrow();
+    });
+
+    test("it mints a domain name and gives it to the sender", async () => {
+      const name = randomString();
+      const alice = Account.generate();
+      await aptos.fundAccount({
+        accountAddress: alice.accountAddress.toString(),
+        amount: 500_000_000,
+      });
+
+      await signAndSubmit(
+        alice,
+        await aptos.registerName({
+          name,
+          expiration: { policy: "domain", years: 1 },
+          sender: alice,
+        }),
+      );
+
+      const owner = await aptos.getOwnerAddress({ name });
+      expect(owner).toEqual(alice.accountAddress.toString());
+    });
+
+    test("it mints a domain name and gives it to the specified address", async () => {
+      const alice = Account.generate();
+      await aptos.fundAccount({
+        accountAddress: alice.accountAddress.toString(),
+        amount: 500_000_000,
+      });
+
+      const bob = Account.generate();
+      await aptos.fundAccount({
+        accountAddress: bob.accountAddress.toString(),
+        amount: 500_000_000,
+      });
+
+      const name = randomString();
+
```

Comment:
Let's followup in a future PR, but we should test some border conditions on the input strings to make sure they work. e.g. -hello hello- he-llo

Reply:
I just added a couple to the `isValidANSName` test